ionoscloud/data_source_cube_server.go

```diff
@@ -0,0 +1,446 @@
+			},
+			"cpu_family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"boot_cdrom": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"boot_volume": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"boot_image": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cdroms": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     cdromsServerDSResource,
+			},
+			"volumes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"availability_zone": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"image_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"image_password": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ssh_keys": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"bus": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"licence_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"cpu_hot_plug": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"ram_hot_plug": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"nic_hot_plug": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"nic_hot_unplug": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"disc_virtio_hot_plug": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"disc_virtio_hot_unplug": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"device_number": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"pci_slot": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"backup_unit_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"user_data": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"boot_server": {
+							Type:        schema.TypeString,
+							Description: "The UUID of the attached server.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"nics": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     nicServerDSResource,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func setCubeServerData(d *schema.ResourceData, server *ionoscloud.Server, token *ionoscloud.Token) error {
```

Comment:
maybe make a setServerData that has the common stuff, then a setEnterpriseData and setCubeData that have only the cube and enterprise parts. That way we don't get duplicated code.